### PR TITLE
[Merged by Bors] - added deployment teardown and removed warnings

### PR DESCRIPTION
**Pull request by AmitShaul**
## Motivation
remove deployment before removing node pool

## Changes
added deployment teardown and removed warnings

### CODE REVIEW

File: tests/app_engine/main.py [hunk 1]
```diff
@@ -27,7 +35,8 @@ def validate_params(request_json):
 
 @app.route('/', methods=['POST'])
 def teardown():
-    print(f"teardown")
+    print(f"starting teardown")
```
Comment (y0sher) [inline tests/app_engine/main.py]: isn't `f` used only for formatting ?

File: tests/app_engine/resources/k8s_handler.py [hunk 2]
```diff
@@ -53,3 +53,42 @@ def remove_clusterrole_binding(project_id, cluster_id, zone, shipper_name, crb_n
...
+    :param cluster_id:
+    :param zone:
+    :param namespace: string, namespace name
+    :param deps: list, a list of all deployments to be deleted under the given namespace
+    :param keywords: list, in case deps wasn't supplied, all deployments who's in their name one of the items in keyword
+                        will be listed for deletion
+    :return:
+    """
+    if not deps:
+        deps = list_namespace_deployments(project_id, cluster_id, zone, namespace, keywords)
```
Comment (y0sher) [inline tests/app_engine/resources/k8s_handler.py:81]: if keywords is `None` we get all deps?

**Review by y0sher — APPROVED** [review]

File: tests/app_engine/resources/k8s_handler.py [hunk 3]
```diff
@@ -53,3 +53,42 @@ def remove_clusterrole_binding(project_id, cluster_id, zone, shipper_name, crb_n
...
+    remove deployments in namespace
+    :param project_id:
+    :param cluster_id:
+    :param zone:
+    :param namespace: string, namespace name
+    :param deps: list, a list of all deployments to be deleted under the given namespace
+    :param keywords: list, in case deps wasn't supplied, all deployments who's in their name one of the items in keyword
+                        will be listed for deletion
+    :return:
+    """
```
Comment (noamnelke) [inline tests/app_engine/resources/k8s_handler.py]: I find it weird that we add a docstring and then leave it half-empty...

File: tests/app_engine/main.py [hunk 4]
```diff
@@ -43,16 +52,22 @@ def teardown():
     namespace = payload_dict['namespace']
     project_id = payload_dict["project_id"]
     print(f"starting teardown process for {namespace}")
+    # delete client and bootstrap deployments
+    try:
+        remove_client_deployments(
+            payload_dict["project_id"], payload_dict["cluster_name"], payload_dict["node_pool_zone"], namespace)
+    except Exception as e:
+        print(f"got an exception while trying to delete deployments under {namespace}:\n{e}")
```
Comment (noamnelke) [inline tests/app_engine/main.py:60]: If the deployment removal failed, we really want to continue? Is there a chance that removing the node pool will succeed?

File: tests/app_engine/resources/k8s_handler.py (hunk 2 above)
Comment (AmitShaul) [reply tests/app_engine/resources/k8s_handler.py:81]: and you didn't mention which deployment you'd like to delete

File: tests/app_engine/main.py (hunk 4 above)
Comment (AmitShaul) [reply tests/app_engine/main.py:60]: we didn't remove deployments so far so, it won't affect the node pool deletion

File: tests/app_engine/resources/k8s_handler.py (hunk 3 above)
Comment (AmitShaul) [reply tests/app_engine/resources/k8s_handler.py]: those can be derived by the param name though I can add it

File: tests/app_engine/main.py (hunk 1 above)
Comment (AmitShaul) [reply tests/app_engine/main.py]: removed

File: tests/app_engine/resources/k8s_handler.py (hunk 3 above)
Comment (noamnelke) [reply tests/app_engine/resources/k8s_handler.py]: what's the point of using this format? I think that it's supposed to be self-documenting code, but empty fields will make the format look broken.

I agree that the names make the meaning clear without documentation, but adding half the documentation just seems weird to me.

**Review by noamnelke — APPROVED** [review]

**Comment by AmitShaul** [comment]
bors merge

**Comment by AmitShaul** [comment]
bors try

**Comment by AmitShaul** [comment]
bors merge

**Comment by AmitShaul** [comment]
bors merge

**Pull request closed without merging**